components/Map/CurrentOutivityMarker.js

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from "react";
+import { useMapEvents } from "react-leaflet/hooks";
+import { Marker, Popup } from "react-leaflet";
+import Link from "next/link";
+import styled from "styled-components";
+
+export default function CurrentOutivityMarker({ outivity, position }) {
+  const [currentPosition, setCurrentPosition] = useState(position);
+
+  const map = useMapEvents({
+    locationfound: (event) => {
+      setCurrentPosition(event.latlng);
+      map.flyTo(event.latlng, map.getZoom());
+    },
+  });
+
+  useEffect(() => {
+    if (position) {
+      setCurrentPosition(position);
+      map.flyTo(position, map.getZoom());
+    }
+  }, [position, map]);
+
+  return (
+    currentPosition && (
```

Comment:
Der Marker sollte doch lieber die Position von eurer Outivity anzeigen, oder?
```suggestion
    position && (
```